rewriter/Prop.cc

```diff
@@ -604,6 +604,17 @@ void Prop::run(core::MutableContext ctx, ast::ClassDef *klass) {
             continue;
         }
 
+        if (wantTypedInitialize(syntacticSuperClass)) {
+            auto it = absl::c_find_if(props, [&propInfo](auto &existing) { return existing.name == propInfo->name; });
+            if (it != props.end()) {
```

Comment:
This is accidentally quadratic in the number of props, right?  (I would expect the number of props to usually be small, but we do have folks creating large numbers of props on some classes.)

I guess moving the detection out of the loops is hard, because we're updating `replaceNodes` and `props` with the results, and we want to avoid those modifications.  Can we instead:

1. Just accumulate things into `props` normally during the loop;
2. Check `props` for duplicates after the loop and remove duplicates from it;
3. Construct `replaceNodes` with the remaining props?

Reply:
Summarizing from slack:

My thinking had been that the prop data we're accumulating would probably be in cache, and that querying it here would be cheap (even though it's quadratic). However, `PropInfo` is quite big, and that assumption probably doesn't hold.

We discussed using a map of `NameRef` to `props` index as an alternative here, as presence in that map would allow us quick access to the original definition for the purposes of error reporting. I like that approach, and will implement it here :)